### PR TITLE
add enums for mavlink ftp service

### PR DESCRIPTION
**Pull request by muexxl**
as aligned on discord

### CODE REVIEW

**Comment by hamishwillee** [comment]
Thanks. Can you format this correctly. There is a script to do this in mavlink/mavlink as per the error log. On linux all you have to do is call `./scripts/format_xml.sh ./message_definitions/v1.0/common.xml`

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -4676,6 +4676,99 @@
       <entry value="1" name="CAN_FILTER_ADD"/>
       <entry value="2" name="CAN_FILTER_REMOVE"/>
     </enum>
+    <enum name="MAV_FTP_ERR">
+			<description>MAV FTP Error Codes</description>
+			<entry value="0" name="MAV_FTP_ERR_NONE">
+				<description>No error </description>
+			</entry>
+			<entry value="1" name="MAV_FTP_ERR_FAIL">
+				<description>Unknown failure</description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:4685]: I think we need to map each of these to the original names in [the FTP doc](https://mavlink.io/en/services/ftp.html#error_codes). The reason is that these names are used in the original FTP specification, and we want to match that. Proposal is to use format - original text, then our description. 
So:
```suggestion
				<description>Fail: Unknown failure</description>
```

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -4676,6 +4676,100 @@
...
+    <enum name="MAV_FTP_ERR">
+			<description>MAV FTP error codes (https://mavlink.io/en/services/ftp.html)</description>
+			<entry value="0" name="MAV_FTP_ERR_NONE">
+				<description>No error </description>
+			</entry>
+			<entry value="1" name="MAV_FTP_ERR_FAIL">
+				<description>Unknown failure</description>
+			</entry>
+			<entry value="2" name="MAV_FTP_ERR_FAILERRNO">
+				<description> Command failed, Err number sent back in PayloadHeader.data[1]. This is a file-system error number understood by the server operating system </description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:4688]:
```suggestion
				<description>FailErrno: Command failed, Err number sent back in PayloadHeader.data[1].
				This is a file-system error number understood by the server operating system </description>
```

File: message_definitions/v1.0/common.xml [hunk 3]
```diff
@@ -4676,6 +4676,100 @@
...
+			</entry>
+			<entry value="3" name="MAV_FTP_ERR_INVALIDDATASIZE">
+				<description> Payload size is invalid </description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:4691]:
```suggestion
				<description>InvalidDataSize: Payload size is invalid </description>
```

File: message_definitions/v1.0/common.xml [hunk 4]
```diff
@@ -4676,6 +4676,100 @@
...
+			</entry>
+			<entry value="10" name="MAV_FTP_ERR_FILENOTFOUND">
+				<description> File/directory not found </description>
+			</entry>
+		</enum>
+		<enum name="MAV_FTP_OPCODE">
+			<description>MAV FTP opcodes (https://mavlink.io/en/services/ftp.html)
+			</description>
+			<entry value="0" name="MAV_FTP_OPCODE_NONE">
+				<description>Ignored, always ACKed</description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:4719]:
```suggestion
				<description>None. Ignored, always ACKed</description>
```

File: message_definitions/v1.0/common.xml [hunk 5]
```diff
@@ -4676,6 +4676,100 @@
...
+			</entry>
+			<entry value="1" name="MAV_FTP_OPCODE_TERMINATESESSION">
+				<description>Terminates open Read session</description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:4722]:
```suggestion
				<description>TerminateSession. Terminates open Read session</description>
```

File: message_definitions/v1.0/common.xml [hunk 6]
```diff
@@ -4676,6 +4676,100 @@
...
+			</entry>
+			<entry value="2" name="MAV_FTP_OPCODE_RESETSESSION">
+				<description>Terminates all open read sessions</description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:4725]:
```suggestion
				<description>ResetSessions. Terminates all open read sessions</description>
```

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (muexxl) [reply message_definitions/v1.0/common.xml]: done as requested. 
The changes are in commit [9de0f89](https://github.com/mavlink/mavlink/pull/1801/commits/9de0f89fad651e6a6a669348fd0edde896c09d58)

**Comment by hamishwillee** [comment]
Thanks very much @muexxl - I've made minor further consistency changes.

@auturgy @julianoes This adds FTP opcodes and error numbers as enums to common.xml. 

This was suggested by @peterbarker because it would allow the FTP protocol implementations to use documented values rather than magic numbers for the opcodes. That's a bit safter and more robust. If this is merged, I will add a link to these in the devguide.

Can you please confirm this is all OK to you?

**Comment by muexxl** [comment]
Thank you @hamishwillee  for helping me to refine my first pull request .. ever :)

**Review by hamishwillee — APPROVED** [review]
This was approved in the last dev call (modulo errors being fixed). Thanks @muexxl !

**Pull request merged**